dialogue-apache-hc4-client/src/main/java/com/palantir/dialogue/hc4/ApacheHttpClientChannels.java

```diff
@@ -235,6 +171,127 @@ public String toString() {
         }
     }
 
+    public static CloseableClientBuilder closeableClientBuilder() {
+        return new CloseableClientBuilder();
+    }
+
+    public static final class CloseableClientBuilder {
+
+        @Nullable
+        private ClientConfiguration clientConfiguration;
+
+        @Nullable
+        private String clientName;
+
+        @Nullable
+        private ExecutorService executor;
+
+        private CloseableClientBuilder() {}
+
+        public CloseableClientBuilder clientConfiguration(ClientConfiguration value) {
+            this.clientConfiguration = Preconditions.checkNotNull(value, "ClientConfiguration is required");
+            return this;
+        }
+
+        public CloseableClientBuilder clientName(String value) {
+            this.clientName = Preconditions.checkNotNull(value, "ClientConfiguration is required");
+            return this;
+        }
+
+        public CloseableClientBuilder clientName(ExecutorService value) {
```

Comment:
`s/clientName/executor/`

Also can we describe in javadoc what this executor will get used for?

Reply:
Updated with docs